src/Kernel/client/plotting.ts

```diff
@@ -27,6 +27,272 @@ export interface DisplayableState {
+            return `|${bitstring}⟩`;
+        }), //basis state labels
+        datasets: [
+            {
+                data: Array.from(Array(newCount).keys()).map(idx => {
+                    return (amps[idx].Real);
+                }),
+                backgroundColor: "#5390d9",
+                borderColor: "#5390d9",
+                label: "Real"
+            },
+            {
+                data: Array.from(Array(newCount).keys()).map(idx => {
+                    return (amps[idx].Imag);
+                }),
+                backgroundColor: "#48bfe3",
+                borderColor: "#48bfe3",
+                label: "Imag"
+            }
+        ],
+    };
+    chart.options.legend = {
+        display: false,
+    };
+    chart.options.scales = {
+        xAxes: [{
+            scaleLabel: {
+                display: true,
+                labelString: 'Basis States'
+            },
+            ticks: {
+                maxRotation: 0,
+                minRotation: 0
+            }
+        }],
+        yAxes: [{
+            scaleLabel: {
+                display: true,
+                labelString: 'Meas. Probability'
+            },
+            ticks: {
+                beginAtZero: true,
+                suggestedMax: 1,
+                suggestedMin: -1
+            }
+        }]
+    };
+
+    chart.update();
+}
+
+export function createNewCanvas(
+    parentNode: HTMLElement, initalState?: DisplayableState | null
+): { canvas: HTMLCanvasElement, chart: ChartJs } {
+    let canvas = document.createElement("canvas");
+    let measurementHistogram = new Chart(canvas, {
+        type: 'bar',
+        options: {
+            responsive: false
+        }
+    });
+
+    if (initalState !== null && initalState !== undefined) {
+        updateWithAmplitudeSquaredData(measurementHistogram, initalState);
+    }
+
+    parentNode.appendChild(canvas);
+
+    return {
+        canvas: canvas,
+        chart: measurementHistogram
+    };
+}
+
+export function attachDumpMachineToolbar(element: HTMLCanvasElement, chart: ChartJs, state: DisplayableState) {
+    //create legend and append to state div
+    let legendDiv = document.createElement("div");
+    legendDiv.id = "legendDiv";
+   
+    let state_div = state.div_id;
+    let div = document.getElementById(state_div);
+    div.appendChild(legendDiv);
+
+    //style the legend
+    document.getElementById("legendDiv").style.border = "thin solid #000000";
+    document.getElementById("legendDiv").style.textAlign = "center";
+
+    let legendTitle = document.createElement("p");
+    legendTitle.innerHTML = "Select Graph";
+
+
+    //create buttons
+    let button = document.createElement("button");
+    button.innerHTML = "amplitude^2";
+
+    let button2 = document.createElement("button");
+    button2.innerHTML = "amplitude/phase";
+
+    var button3 = document.createElement("button");
+    button3.innerHTML = "real/imag";
```

Comment:
Minor comment, but perhaps some more descriptive names than `button`, `button2`, and `button3` would be useful. Perhaps `buttonAmplitudeSquared`, `buttonAmplitudePhase`, and `buttonRealImag`?